File: app/src/main/java/com/designpatterns/demo/abstractfactory/java/MouldProductA.java
package com.designpatterns.demo.abstractfactory.java;

/**
 * 创建具体产品类（集成抽象产品类），定义生成的具体产品
 *
 * @author zhangzhiguo
 * @version 1.0.0
 * @email: 810903987@qq.com
 * @date 2018/5/18 下午1:44
 * @project_name DesignPatternsDemo
 */
public class MouldProductA extends MouldProduct {
    @Override
    public void Show() {
        System.out.println("生成出模具产品A");
    }
}

File: app/src/main/java/com/designpatterns/demo/creational/abstractfactory/kotlin/MouldProductA.kt
package com.designpatterns.demo.creational.abstractfactory.kotlin

/**
 * 创建具体产品类（集成抽象产品类），定义生成的具体产品
 *
 * @author  zhangzhiguo
 * @email: 810903987@qq.com
 * @date 2018/5/21 上午10:00
 * @project_name DesignPatternsDemo
 * @version 1.0.0
 *
 */
class MouldProductA: MouldProduct(){
    override fun Show() {

        println("生成出模具产品A")

    }
}